yellowbrick/contrib/scatter.py

```diff
@@ -215,6 +215,12 @@ def fit(self, X, y=None, **kwargs):
         """
         _, ncols = X.shape
 
+        # NOTE: Do not call super for this class, it conflicts with the fit.
```

Comment:
I can take a look at those classes to see why they do not call super. I feel that if a subclass does not call super then that means that it might be poorly subclassed - e.g. it is using some functionality of the superclass and not other functionality. As we move forward if this is the case we should consider:

1. Breaking the superclass into two classes or a mixin and extending the inheritance tree
2. Extensively commenting why we don't call super (as in the note above)

Otherwise:

All of our visualizers should call super. 

Not calling super is a code smell and it can definitely break our code if we add functionality to the superclass but that is not propagated down to all of our visualizers! And you're right, it gets even harder the larger Yellowbrick gets and the farther we are from the original code.

Reply:
Ah I see - I was mistaken; all of the visualizers you listed do call super, but they don't say why they call super either at the top or at the bottom of the method. I agree we can start doing a better job of that and when I'm working somewhere where I notice that isn't specifically documented, I will do so!